protocol/storage/client.go

```diff
@@ -70,23 +66,29 @@ type clientPorcelainAPI interface {
 
 // Client is used to make deals directly with storage miners.
 type Client struct {
-	node clientNode
-	api  clientPorcelainAPI
+	api                 clientPorcelainAPI
+	blockTime           time.Duration
+	host                host.Host
+	log                 logging.EventLogger
+	ProtocolRequestFunc func(ctx context.Context, protocol protocol.ID, peer peer.ID, host host.Host, request interface{}, response interface{}) error
 }
 
 // NewClient creates a new storage client.
-func NewClient(nd clientNode, api clientPorcelainAPI) (*Client, error) {
+func NewClient(blockTime time.Duration, host host.Host, api clientPorcelainAPI) *Client {
 	smc := &Client{
-		node: nd,
-		api:  api,
+		api:       api,
+		blockTime: blockTime,
+		host:      host,
+		log:       logging.Logger("storage/client"),
 	}
-	return smc, nil
+	smc.ProtocolRequestFunc = MakeProtocolRequest
```

Comment:
curious why set this outside the above?

Reply:
good catch, it was originally not a free function and I forgot to move it back in.